### ISSUE TITLE
[Syphilis] [SPIKE] Support customized backend results label/SNOMED by disease or by device

### ISSUE THREAD

**Issue by jayna-SkylightDigital** [CDCgov/prime-simplereport#7395]
## Why should we do this?
- (current experience) Currently SimpleReport sends the same results to public health departments, regardless of the format used by the device: positive, negative, or inconclusive.
- (why we want to make this change) Results sent to public health departments should reflect the label format of the device used. While "positive" and "negative" is commonly used when referring to syphilis test results, because treponemal tests confirm the presence of antibodies and can't on their own confirm a syphilis diagnosis, it's more correct to say tests are **reactive** or **non-reactive**. LAC currently uses the following language in their own reporting:

  - Positive/Reactive
  - Negative/Nonreactive
  - Inconclusive

## User story
As jurisdiction receiving test results captured by SimpleReport I want to receive the results in the correct format so that I can rely on those results for my analysis.

## Action Requested
- Investigate how results are currently being transmitted to public health departments.
- Determine how best to proceed to provide results in the format that public health departments expect to receive them: positive/negative, reactive, non-reactive, detected/not-detected, etc.

## Acceptance Criteria
- Creation of additional tickets that will be used for implementation of the chosen solution.

**Review**
-   [ ] Reviewed with design and/or design lead
-   [ ] Reviewed with content strategist for final approval
-   [ ] Reviewed with product and engineering

## Additional Context
- What is the standard?
- Who defines accuracy of results (SNOMED)
- Should frontend reflect what we’re sending via device?
- Laura looked into this for HIV - ticket may be in the backlog (closed)

**Comment by fzhao99** [comment]
**TL;DR:** The LIVD table uses a number of different codes across different devices, but we map all cases of positive / presumptive positive / detected into the SNOMED for detected and negative / not detected into not detected. Since we're already sending data to STLT's in this manner, I'd recommend we just stick with what we have unless explicitly asked.

 If we do want to do something however, we should try pulling SNOMED codes from the LIVD table at the device level in the instances where we do this and then do the mapping on our backend. We'd need to double check with RS to make sure we won't break anything doing this. We could do something similar manually for the devices we're adding for the new STI's.

**More info**

The following are some of the relevant codes that might be relevant for our use case.

- Presumptive positive - 720735008
- Reactive - 11214006
- Detected - 260373001 (what we currently use, coded both the frontend / backend / in the db as “positive”)
- Positive - 10828004
     - Should note here that a positive result on RT-PCR COVID/flu devices match the clinical confirmed cased definition, so some devices on the LIVD table / in the app should technically use this value even though we code it simply as “detected”  

- Not detected - 260415000 - (what we currently use, frontend /backend “negative”)
- Nonreactive - 131194007
- Negative - 260385009
    - Similar story "positive" for our RT-PCR devices

- Invalid result - 455371000124106 - (what we currently use, frontend “inconclusive”, backend “undetermined”)

On the bulk upload flow, we have a map for positive, negative, detected, not detected, and invalid that we use to translate the strings into SNOMED’s. For positive / negative, we do an interpretive mapping into detected / not detected (ie use the SNOMED’s for detected / not detected for positive and negative values). This means we already send the detected / not detected values to STLT's (even when we should technically send the positive / negative SNOMED's) even though users see positive / negative on the frontend. On the single entry flow, we handle this in the TestEventExport object and use the codes for detected / not detected for positive / negative values coming from the frontend. 

Can’t figure out whether there’s a significant difference between reactive / detected and non-reactive / not detected, but in the SNOMED taxonomy, detected / not detective are “finding values” with taxonomical children (anaerobes / pooled specimen), while reactive / non-reactive are top-level “adjectival modifiers”. In practice though, I think these are probably pretty synonymous.

Can’t remember the precise context here, but the fact that we’re technically sending “inaccurate” data to the STLTs already says to me that we should be fine keeping things the way things are unless someone asks.

If someone does ask, next steps would probably be to check in with ReportStream to ensure that tweaking values wouldn’t cause any issues for them. We'd then need to add mapping between the device-level SNOMED's for the "positive" and "negative" cases and then seed that data from the LIVD table or manually.

**Comment by fzhao99** [comment]
Discussed with the eng team and general opinion is that the best action is that we'll just keep the backend mapping as is for syphillis since it'll be "more accurate" once we add the STI's.

Follow up item: @jayna-SkylightDigital @kenieh do yall have opinions on what frontend copy we want to add to the test card? Since we're sending "detected / not detected" would we want to change that anywhere?

@jayna-SkylightDigital pending the answer to the above q, if you could weigh in on whether this is sufficient to close the spike that'd be great!

**Comment by fzhao99** [comment]
documented the writeup [here](https://github.com/CDCgov/prime-simplereport/wiki/Test-Metadata-and-how-we-store-it#a-note-on-how-we-represent-positive--negative--inconclusive-results)

**Comment by fzhao99** [comment]
determined that we'll keep things the way they are with the positive / negative labels and revisit if necessary. Slack thread below for context

![Image](https://github.com/CDCgov/prime-simplereport/assets/29645040/9ca11dea-67e4-4a7d-9c0c-13447f07361d)
